### ISSUE TITLE
When setting options to empty - set value to empty as well

### ISSUE THREAD

**Issue by jsignell** [pyviz/panel#306]
I was pleasantly surprised with the behavior of value when options are changed. The only weirdness I've seen is that when options is set to empty - value should also be set to empty
```python
import panel as pn

select = pn.widgets.Select(options={'a': 1, 'b': 2, 'c': 3}, value=2)
select

select.options.pop('c')
select.param.trigger('options')
print(select.options, select.value)
# {'a': 1, 'b': 2} 2

select.options.pop('b')
select.param.trigger('options')
print(select.options, select.value)
# {'a': 1} 1

select.options.pop('a')
select.param.trigger('options')
print(select.options, select.value)
# {} 1
```

**Comment by jsignell** [comment]
Note that MultiSelect does the right thing:
```python
select = pn.widgets.MultiSelect(options={'a': 1, 'b': 2, 'c': 3}, value=[1, 2])
...
# {'a': 1, 'b': 2} [1, 2]
# {'a': 1} [1]
# {} []
```

**Comment by philippjfr** [comment]
I initially had some issues getting the Select widget to accept the None values but it seems to work fine in my latest commit to https://github.com/pyviz/panel/pull/305.

**Comment by jsignell** [comment]
I just noticed that the behavior isn't consistent when using set vs trigger.

```python
select.options.pop('c')
select.options = select.options
print(select.options, select.value)

select.options.pop('b')
select.options = select.options
print(select.options, select.value)

select.options.pop('a')
select.options = select.options
print(select.options, select.value)

# {'a': 1, 'b': 2} 2
# {'a': 1} 2
# {} 2

# {'a': 1, 'b': 2} [1, 2]
# {'a': 1} [1, 2]
# {} [1, 2]
```
Notably - value is not affected and widget does not re-render

**Comment by philippjfr** [comment]
>I just noticed that the behavior isn't consistent when using set vs trigger.

Looking at this now, quite confused by this.

**Comment by philippjfr** [comment]
Actually it's not totally surprising, by modifying the options inplace you remove param's ability to detect that the contents of the dictionary have changed. Not sure what to do about this but it's not something we can address in panel, we'd have to look at param.

**Comment by jsignell** [comment]
> by modifying the options inplace you remove param's ability to detect that the contents of the dictionary have changed

I guess that makes sense. It's just that I have objects in the options so even updating on an shallow copy doesn't trigger the change. But I agree it's a param issue.

Fixed in #307. Thanks!
